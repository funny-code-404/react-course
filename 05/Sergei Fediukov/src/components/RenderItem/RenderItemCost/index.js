const RenderItemCost = (props) => {
    const { cost } = props
    return (
        <div>
            <p>Costs:</p>
            <ul>
                {cost.Wood && <li>Wood - <span>{cost.Wood}</span></li>}
                {cost.Gold && <li>Gold - <span>{cost.Gold}</span></li>}
                {cost.Food && <li>Food - <span>{cost.Food}</span></li>}
                {cost.info && <li>Cost - <span>{cost.info}</span></li>}
                {cost.Stone && <li>Stone - <span>{cost.Stone}</span></li>}
            </ul>
        </div>
    )
}
export default RenderItemCost