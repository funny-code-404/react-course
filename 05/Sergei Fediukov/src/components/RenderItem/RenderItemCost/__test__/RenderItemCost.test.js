import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import RenderItemCost from '..'

configure({ adapter: new Adapter() });

describe('Should render correctly', () => {

    const props = {
        cost: {
            Wood: 'some wood',
            Food: 'some stone',
            Gold: 'some gold',
            Stone: 'some stone',
            info: 'some info'
        }
    }
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<RenderItemCost {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
})