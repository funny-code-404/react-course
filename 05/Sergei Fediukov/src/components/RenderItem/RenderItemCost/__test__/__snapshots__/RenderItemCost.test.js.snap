// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly to Match Snapshot 1`] = `
<div>
  <p>
    Costs:
  </p>
  <ul>
    <li>
      Wood - 
      <span>
        some wood
      </span>
    </li>
    <li>
      Gold - 
      <span>
        some gold
      </span>
    </li>
    <li>
      Food - 
      <span>
        some stone
      </span>
    </li>
    <li>
      Stone - 
      <span>
        some stone
      </span>
    </li>
    <li>
      Cost - 
      <span>
        some info
      </span>
    </li>
  </ul>
</div>
`;
